internal/pullrequest/queue/collect_result.go

```diff
@@ -105,14 +105,19 @@ func (c *controller) sendPullRequestQueueReport(ctx context.Context, prQueue *s2
 }
 
 func (c *controller) isTearDownDurationCriteriaMet(criteria s2hv1.PullRequestTearDownDurationCriteria,
-	result s2hv1.PullRequestQueueResult) bool {
+	prq *s2hv1.PullRequestQueue) bool {
+	if (prq.Spec.IsPRTriggerFailed != nil && *prq.Spec.IsPRTriggerFailed) ||
+		!prq.Status.IsConditionTrue(s2hv1.PullRequestQueueCondDeployed) {
```

Comment:
I don't think that we can check from this condition `s2hv1.PullRequestQueueCondDeployed`, as checked in the code this will be set only the deployment passes https://github.com/agoda-com/samsahai/blob/69222a7178d092cb564d30bb5e358a78910b57a8/internal/pullrequest/queue/queue_deployment.go#L127-L132

might need to check from condition `QueueDeployStarted`
https://github.com/agoda-com/samsahai/blob/579c7b5cc08c1bdf565e74c0e64b204320016475/api/v1/queue_types.go#L232

from `DeploymentQueue` instead https://github.com/agoda-com/samsahai/blob/69222a7178d092cb564d30bb5e358a78910b57a8/api/v1/pullrequestqueue_types.go#L182

Reply:
If it be true only when deployed successfully, doesn't it mean that when the deployment doesn't succeed, it will be either false or undefined, which will make `!prq.Status.IsConditionTrue(s2hv1.PullRequestQueueCondDeployed) == true` anyway?

Reply:
refer to these lines
https://github.com/agoda-com/samsahai/blob/eb27143cc63a3211b02753338c44ae399ad71c76/api/v1/pullrequestqueue_types.go#L212-L219

Reply:
correct, it will be set as false if it fails 
https://github.com/agoda-com/samsahai/blob/69222a7178d092cb564d30bb5e358a78910b57a8/internal/pullrequest/queue/queue_deployment.go#L138-L140

```
!prq.Status.IsConditionTrue(s2hv1.PullRequestQueueCondDeployed) == true
```
and this is correct

but if we check this case, so does it mean that whenever the deployment fails, we will not keep the namespace even the criteria is `failure`?

Reply:
Yeah. Also, I'm not sure whether it's useful to keep the environment if the deployment failed. Can the user still conduct some manual test if the deployment failed?

Reply:
my concern is if the deployment starts and fails, then `PullRequestQueueCondDeployed` will be `false` from the code [above](https://github.com/agoda-com/samsahai/blob/69222a7178d092cb564d30bb5e358a78910b57a8/internal/pullrequest/queue/queue_deployment.go#L138-L140)

then `!prq.Status.IsConditionTrue(s2hv1.PullRequestQueueCondDeployed)` also be `true`, then the `isTearDownDurationCriteriaMet` function will return `false` without checking the criteria and the pull request namespace will be destroyed.

Please correct me if I'm wrong.

Reply:
so my suggestion is instead of checking `PullRequestQueueCondDeployed` condition which is `true` only it's success, checking from `QueueDeployStarted` might be better

Reply:
**Conclusion:**
We will check the criteria only if the deployment passes for now since if it fails, users could see from logs instead.
Then, this condition is already correct.